### PR TITLE
Develop feature

### PR DESCRIPTION
**Pull request by AlejandroNo4**
This project was made for learning proposes, to practice the way of creating forms with Rails. It was for practice and understand the way data is structured and sent to the controller.

### CODE REVIEW

**Review by LamiaSristy — CHANGES_REQUESTED** [review]
# Status: Change Required. ♻️

Hello @AlejandroNo4. 
**I can assume a lot of effort and dedication needs for coding, and all are here present in your project.**

## The highlights: 
* The `linters` set up correctly.
*  The create and edit user actions are working fine.
*  Great job with error messages for fail validations.

## What to do next: 

**Please make sure all the required point added to get approval**

**[REQUIRED]** 👇 

- [x] In the `README` file, kindly add instructions for cloning and running the project locally, ex: run `bundle install &&  yarn  && rails db:migrate && rails s`. This will help the user to easily clone your project.

- [x] Kindly give your PR a meaningful title with a proper description of the project and a little more descriptive commit messages, so that your PR is professional.

- [x]  In the `UsersController`, `update` action, kindly change the ` redirect_to @user` to  `redirect_to edit_user_path`, as currently you created only :new, :create, :edit, :update actions. So you need to redirect to  `edit_user_path` if user is updated.

- [x] In the views, kindly use the tag `form_with`, as it is given as instructions in the [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms) project.  See my attached image also 👇 
odin
##

**Your feature branch's name is not appropriate, consider not using the word develop in the naming of feature-branch, this will be ensuring your repository is more professional**

Please make the changes and ask for a re-review.


Feel free to reach me on slack @lamiasristy 🙋

Keep up the spirit always. Happy Coding 💻

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  def update
+    @user = User.find(params[:id])
+
+    if @user.update(user_params)
+      redirect_to @user
+    else
+      render :edit
+    end
```
Comment (LamiaSristy) [inline app/controllers/users_controller.rb]: Change the redirection after updating the user, `redirect_to edit_user_path`

File: app/views/users/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  <div><%= message %></div>
+  <% end %>
+  <br /><br />
+  <%= f.label :password, "Password:" %><br />
+  <%= f.password_field :password %>
+  <% @user.errors.full_messages_for(:password).each do |message| %>
+  <div><%= message %></div>
+  <% end %>
+  <br /><br />
+  <%= f.submit %>
```
Comment (LamiaSristy) [inline app/views/users/edit.html.erb]: Use `#form_with` here as it was instructed in the [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms) project.

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  <% end %>
+  <br /><br />
+  <%= f.label :password, "Password:" %><br />
+  <%= f.password_field :password, placeholder: "Password" %>
+  <% @user.errors.full_messages_for(:password).each do |message| %>
+  <div><%= message %></div>
+  <% end %>
+  <br /><br />
+  <%= f.submit %>
+<% end %>
```
Comment (LamiaSristy) [inline app/views/users/new.html.erb]: Use `#form_with` here as it was instructed in the [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms) project.

**Comment by AlejandroNo4** [comment]
I had problems in the past changing the name of the branch. I decided to leave the name as it is, Investigate more about how to do it, and avoid doing it in the future. 
Also, I will try to make better commits.

Thank you!

**Review by mke2111 — APPROVED** [review]
# **🎉 STATUS: APPROVED 🎉** 

## Hey @AlejandroNo4, 👋🏽 
 **Congratulations 🥳 🥂 . Your app meets the minimum requirements for approval.**

- [x] I'm approving this project since you had initially used the `form_for` in the edit.html.erb.
- [x] [OPTIONAL] In the readme file. Add all commands needed to get the app ready before starting the server. Such as `bundle install` and `rails db:migrate`.

**You may now merge to the develop branch and submit a project completion form. 👍🏽** 

## Cheers!! 🍹 

## Reach me on slack for a chitchat 🆘  **Roy Mukuye**

**Pull request merged**